scripts/rewrite-decl-statements-for-closure-test.js

```diff
@@ -95,14 +95,47 @@ function transform(srcFile, rootDir) {
   });
 
   traverse(ast, {
-    'ImportDeclaration|ExportNamedDeclaration'({node}) {
-      if (node.source) {
-        rewriteDeclarationSource(node, srcFile, rootDir);
+    ExportNamedDeclaration(path) {
+      const properties = [];
+      path.node.specifiers.forEach((specifier) => {
+        properties.push(t.objectProperty(specifier.exported, specifier.exported, false, true, []));
+      });
+      const right = t.objectExpression(properties);
+      const expression = t.assignmentExpression('=', t.identifier('exports'), right);
+      path.replaceWith(t.expressionStatement(expression));
+    },
+  });
+
+  traverse(ast, {
+    ExportDefaultDeclaration(path) {
+      const expression = t.assignmentExpression('=', t.identifier('exports'), path.node.declaration);
+      path.replaceWith(t.expressionStatement(expression));
+    },
+  });
+
+  traverse(ast, {
+    ImportDeclaration(path) {
+      const callee = t.memberExpression(t.identifier('goog'), t.identifier('require'), false);
+      const packageStr = rewriteDeclarationSource(path.node, srcFile, rootDir);
+      const callExpression = t.callExpression(callee, [t.stringLiteral(packageStr)]);
+
+      let variableDeclaratorId;
+      if (path.node.specifiers.length > 1) {
```

Comment:
I figured out a possible solution to the problem of requiring globbing in place of referencing a single export. You're right, it seems like the AST doesn't reflect the `{` and `}` in the specifier ever, but it does reflect the source locations...

```js
        if (path.node.specifiers.length > 1 ||
          (src.substr(path.node.specifiers[0].start - 1, 1) === '{' && src.substr(path.node.specifiers[0].end, 1) === '}')) {
```

Reply:
Yay! Works!